Add encoding and decoding support for PathOperation

graouts opened the pull request:
#### fd75c72c681f7918ffcd26ecdb52f6a72d8ed423
<pre>
Add encoding and decoding support for PathOperation
<a href="https://bugs.webkit.org/show_bug.cgi?id=249234">https://bugs.webkit.org/show_bug.cgi?id=249234</a>

Reviewed by Alex Christensen.

Straightforward encoding work except for ReferencePathOperation where we resolve the Path
and store it.

* Source/WebCore/rendering/PathOperation.cpp:
(WebCore::ReferencePathOperation::create):
(WebCore::ReferencePathOperation::ReferencePathOperation):
(WebCore::ReferencePathOperation::path const):
(WebCore::RayPathOperation::create):
(WebCore::ReferencePathOperation::getPath const): Deleted.
* Source/WebCore/rendering/PathOperation.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/257832@main">https://commits.webkit.org/257832@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1939b666a970d1b0a6ab7791e350d92980933505

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/100093 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/9260 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/33167 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/109434 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/169668 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/104090 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/10143 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/86762 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/92531 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/107324 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/105862 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/7666 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/90955 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34377 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/89572 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/22349 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/3039 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/23864 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/3004 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Big-Sur-Debug-WK2-Tests-EWS "Waiting to run tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/9136 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/43345 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/4850 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2768 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/rendering/PathOperation.cpp:
```diff
@@ -48,18 +52,32 @@ ReferencePathOperation::ReferencePathOperation(const String& url, const AtomStri
+
     if (!is<SVGPathElement>(m_element.get()) && !is<SVGGeometryElement>(m_element.get()))
         return std::nullopt;
     return pathFromGraphicsElement(m_element.get());
```
Since we're adding m_path, do we want to cache the result of this for if we query the path twice?

graouts replied:
@nmoucht can we compute the path upfront?

nmoucht replied:
For ReferencePathOperation yes it can since it is just computed based off of the SVGElement.

graouts replied:
I've updated the patch to set `m_path` in the constructor so that it's computed upfront.

graouts committed:
Add encoding and decoding support for PathOperation
https://bugs.webkit.org/show_bug.cgi?id=249234

Reviewed by Alex Christensen.

Straightforward encoding work except for ReferencePathOperation where we resolve the Path
and store it.

* Source/WebCore/rendering/PathOperation.cpp:
(WebCore::ReferencePathOperation::create):
(WebCore::ReferencePathOperation::ReferencePathOperation):
(WebCore::ReferencePathOperation::path const):
(WebCore::RayPathOperation::create):
(WebCore::ReferencePathOperation::getPath const): Deleted.
* Source/WebCore/rendering/PathOperation.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/257832@main
